pyro/distributions/transforms/spline.py

```diff
@@ -0,0 +1,352 @@
+
+        logabsdet = torch.log(derivative_numerator) - 2 * torch.log(torch.abs(denominator))
+
+    # Apply the identity function outside the bounding box
+    outputs[outside_interval_mask] = inputs[outside_interval_mask]
+    logabsdet[outside_interval_mask] = 0.0
+    return outputs, logabsdet
+
+
+class SplineLayer(nn.Module):
+    """
+    Helper class to manage learnable spline. One could imagine this as a standard layer in PyTorch...
+    """
+
+    def __init__(self, input_dim, count_bins=8, bound=3., order='linear'):
+        super().__init__()
+
+        self.input_dim = input_dim
+        self.order = order
+
+        # K rational quadratic segments or 2K rational linear segments...
+        self.count_bins = count_bins
+
+        # ...on [-B, B] x [-B, B]
+        self.bound = bound
+
+        # Parameters for each dimension
+        # TODO: What should the initialization scheme be?
+        self.unnormalized_widths = nn.Parameter(torch.randn(self.input_dim, self.count_bins))
+        self.unnormalized_heights = nn.Parameter(torch.randn(self.input_dim, self.count_bins))
+        self.unnormalized_derivatives = nn.Parameter(torch.randn(self.input_dim, self.count_bins - 1))
+
+        # Rational linear splines have additional lambda parameters
+        if self.order == "linear":
+            self.unnormalized_lambdas = nn.Parameter(torch.rand(self.input_dim, self.count_bins))
+        elif self.order == "quadratic":
+            raise ValueError("Monotonic rational quadratic splines not yet implemented!")
+        else:
+            raise ValueError(
+                "Keyword argument 'order' must be one of ['linear', 'quadratic'], but '{}' was found!".format(
+                    self.order))
+
+    @property
+    def widths(self):
+        # widths, unnormalized_widths ~ (input_dim, num_bins)
+        return F.softmax(self.unnormalized_widths, dim=-1)
+
+    @property
+    def heights(self):
+        # heights, unnormalized_heights ~ (input_dim, num_bins)
+        return F.softmax(self.unnormalized_heights, dim=-1)
+
+    @property
+    def derivatives(self):
+        # derivatives, unnormalized_derivatives ~ (input_dim, num_bins-1)
+        return F.softplus(self.unnormalized_derivatives)
+
+    @property
+    def lambdas(self):
+        # lambdas, unnormalized_lambdas ~ (input_dim, num_bins)
+        return torch.sigmoid(self.unnormalized_lambdas)
+
+    def __call__(self, x, jacobian=False, **kwargs):
+        y, log_detJ = monotonic_rational_spline(
+            x,
+            self.widths,
+            self.heights,
+            self.derivatives,
+            self.lambdas,
+            bound=self.bound,
+            **kwargs)
+
+        if not jacobian:
+            return y
+        else:
+            return y, log_detJ
+
+
+@copy_docs_from(TransformModule)
+class Spline(TransformModule):
+    """
+    An implementation of the element-wise rational spline bijections of linear and quadratic order (Durkan et
```

Comment:
nit: can you try to ensure all docstrings wrap at 80 columns? This is important so that, when users type `help(Spline)` in a jupyter notebook, the docstring fits in the window. We generally try to do this with all dostrings in Pyro.

Reply:
I've done it for this class. I will do it for the other classes in a future PR